cmd/server-rlimit.go

```diff
@@ -75,6 +77,27 @@ func setMaxResources() (err error) {
 		return err
 	}
 
+	// set debug memory limit instead of GOMEMLIMIT env
+	_ = setDebugMemoryLimit(ctx)
+
 	err = sys.SetMaxMemoryLimit(maxLimit, maxLimit)
 	return err
 }
+
+func setDebugMemoryLimit(ctx *cli.Context) error {
+	if ctx == nil {
+		return nil
+	}
+	if ctx.IsSet("memlimit") {
```

Comment:
I do not see memlimit flag defined anywhere, you need to define it ServerFlags in server-main.go